libs/error-stack/src/compat.rs

```diff
@@ -13,13 +13,12 @@ mod eyre;
 
 /// Compatibility trait to convert from external libraries to [`Report`].
 ///
-/// *Note*: It's not possible to implement [`IntoReport`] or [`Context`] on other error libraries'
-/// types as both traits have blanket implementation relying on [`Error`]. Thus, implementing either
-/// trait would violate the orphan rule; the upstream crate could implement [`Error`] and this would
-/// imply an implementation for [`IntoReport`]/[`Context`].
+/// *Note*: It's not possible to implement [`Context`] on other error libraries' types as the trait
```

Comment:
Do we maybe want to make this a heading instead?

Reply:
This is the second sentence (and only description) on a trait that is only used for compatibility. I would leave it as it is. I, however, generally improved the documentation a bit.